api/src/main/java/org/openmrs/module/datafilter/impl/ImplDataFilterListener.java

```diff
@@ -87,7 +87,12 @@ public boolean onEnableFilter(DataFilterContext filterContext) {
 			
 		} else if (filterContext.getFilterName().startsWith(ImplConstants.PROGRAM_BASED_FILTER_NAME_PREFIX)) {
 			Collection<String> userProgramRoleNames = new HashSet();
+			//Add '#####' place holder for non authenticated user
+			userProgramRoleNames.add("#####");
 			Collection<String> allProgramRoleNames = AccessUtil.getAllProgramRoles();
+			//Add '#####' place holder for non authenticated user
+			allProgramRoleNames.add("#####");
+			
```

Comment:
Looks like you always set those placeholders now, so then what about [lines 102-110](https://github.com/openmrs/openmrs-module-datafilter/pull/19/files#diff-9131b4706b019c82d4199ff859a46cca876cdf721da3f24b94c8a24cc6cae614R102-R110), are they still needed then?

Or if you want to keep lines 102-110, should you enclose your change in an `if` that checks that the user is **not** authenticated?

Reply:
>Looks like you always set those placeholders now, so then what about [lines 102-110](https://github.com/openmrs/openmrs-module-datafilter/pull/19/files#diff-9131b4706b019c82d4199ff859a46cca876cdf721da3f24b94c8a24cc6cae614R102-R110), are they still needed then?

Yes the lines are needed. `#####` is set as the default which is later overridden/filtered out if the user is authenticated.

>Or if you want to keep lines 102-110, should you enclose your change in an if that checks that the user is not authenticated?

Let me arrange it then in a straight forward piece of code.

>Also, should the same logic not be added further up for the query involved [here](https://github.com/openmrs/openmrs-module-datafilter/blob/72e27f881a3aa68d3de536d1e4ae28021f004806/api/src/main/java/org/openmrs/module/datafilter/impl/ImplDataFilterListener.java#L86)?

No because that piece is not hit at-least by any api call trying to query the user table.